MapboxCoreNavigation/Feedback.swift

```diff
@@ -3,89 +3,48 @@ import Foundation
-    case incorrectVisual
+    /// Indicates an incorrect visual.
+    case incorrectVisual(subtype: IncorrectVisualSubtype)
 
     /// Indicates confusing voice instruction.
-    case confusingAudio
+    case confusingAudio(subtype: ConfusingAudioSubtype)
 
     /// Indicates an issue with route quality.
-    case routeQuality
+    case routeQuality(subtype: RouteQualitySubtype)
 
     /// Indicates that an illegal route was recommended.
-    case illegalRoute
+    case illegalRoute(subtype: IllegalRouteSubtype)
 
     /// Indicates a road closure was observed.
-    case roadClosure
-
+    case roadClosure(subtype: RoadClosureSubtype)
     
     public var description: String {
         switch self {
         case .general:
             return "general"
-        case .incorrectVisual:
+        case .incorrectVisual(_):
             return "incorrect_visual"
-        case .confusingAudio:
+        case .confusingAudio(_):
             return "confusing_audio"
-        case .routeQuality:
+        case .routeQuality(_):
             return "route_quality"
-        case .illegalRoute:
+        case .illegalRoute(_):
             return "illegal_route"
-        case .roadClosure:
+        case .roadClosure(_):
             return "road_closure"
         }
     }
-
-    /// Returns a corresponding list of `FeedbackSubType`s for a given `FeedbackType`
-    public var subtypes: [FeedbackSubType] {
-        switch self {
-        case .general:
-            return []
-        case .incorrectVisual:
-            return [.turnIconIncorrect,
-                    .streetNameIncorrect,
-                    .instructionUnnecessary,
-                    .instructionMissing,
-                    .maneuverIncorrect,
-                    .exitInfoIncorrect,
-                    .laneGuidanceIncorrect,
-                    .roadKnownByDifferentName]
-        case .confusingAudio:
-            return [.guidanceTooEarly,
-                    .guidanceTooLate,
-                    .pronunciationIncorrect,
-                    .roadNameRepeated]
-        case .routeQuality:
-            return [.routeNonDrivable,
-                    .routeNotPreferred,
-                    .alternativeRouteNotExpected,
-                    .routeIncludedMissingRoads,
-                    .routeHadRoadsTooNarrowToPass]
-        case .illegalRoute:
-            return [.routedDownAOneWay,
-                    .turnWasNotAllowed,
-                    .carsNotAllowedOnStreet,
-                    .turnAtIntersectionUnprotected]
-        case .roadClosure:
-            return [.streetPermanentlyBlockedOff,
-                    .roadMissingFromMap]
-
-        }
-    }
-
-
-
 }
 
-/// Enum used to define the many `FeedbackSubType`s that may be nested under a given `FeedbackType`
-public enum FeedbackSubType: String {
-    /// Incorrect Visual Subtypes
+/// Enum denoting the subtypes of the  `Incorrect Visual` top-level category
+public enum IncorrectVisualSubtype: String {
+    case none
```

Comment:
Instead of an explicit `none` subtype, make the associated value optional when declaring the `FeedbackType` case:

```swift
case incorrectVisual(subtype: IncorrectVisualSubtype?)
```

Reply:
I'd considered this, but the call site looked a little weird ( `.incorrectVisual(subtype: nil)` for example) to me. But i'm not strongly against this approach.

Reply:
If it looks too weird, we can make the argument optional with a default value of `nil`.

Reply:
Addressed.